src/util/misc/projectStroke.ts

```diff
@@ -39,62 +49,216 @@ export const projectStrokeOnPoints = (
+      MDN: https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/stroke-linejoin
+      Spec: https://svgwg.org/svg2-draft/painting.html#StrokeLinejoinProperty
+      Playground to understand how the line joins works: https://hypertolosana.github.io/efficient-webgl-stroking/index.html
+      View the calculated projections here for each of the control points: https://codesandbox.io/s/project-stroke-points-with-context-to-trace-b8jc4j?file=/src/index.js
+    */
+
+    // First we calculate the bisector between the points. Used in round and miter cases
+    // When the stroke is uniform, scaling changes the arrangement of the points, so we have to take it into account
+    const bisector = strokeUniform 
+      ? getBisector(A.multiply(scale), B.multiply(scale), C.multiply(scale)) 
+      : getBisector(A, B, C),
       bisectorVector = bisector.vector,
-      alpha = bisector.angle;
-    let scalar, miterVector;
-    if (strokeLineJoin === StrokeLineJoin.miter) {
-      scalar = -s / Math.sin(alpha / 2);
-      miterVector = new Point(
-        bisectorVector.x * scalar * strokeUniformScalar.x,
-        bisectorVector.y * scalar * strokeUniformScalar.y
-      );
-      if (Math.hypot(miterVector.x, miterVector.y) / s <= strokeMiterLimit) {
-        coords.push(A.add(miterVector));
-        coords.push(A.subtract(miterVector));
+      alpha = Math.abs(bisector.angle); //TODO: check if ABS is really needed. I don't remember why I implemented it this way
+
+    /* MITER 
+      Calculation: the corner is formed by extending the outer edges of the stroke at the tangents of the path segments until they intersect
+      Visually detailed here: https://github.com/fabricjs/fabric.js/issues/8025
+    */
+    if (strokeLineJoin === 'miter') {
+      const scalar = -s / Math.sin(alpha / 2),
+        miterVector = scaleHatVector(bisectorVector, scalar);
+
+      // When two line segments meet at a sharp angle, it is possible for the join to extend,
+      // it is possible for the join to extend far beyond the thickness of the line stroking
+      // the path. The stroke-miterlimit imposes a limit on the extent of the line join.
+      // MDN: https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/stroke-miterlimit
+      // When the stroke is uniform, scaling changes the arrangement of points, this changes the miter-limit
+      let adjustedStrokeMiterLimit;
+      if (strokeUniform) {
+        // TODO: As it is in the MDN, the miterLimit can also be calculated as 1/sin(alpha/2) , I believe it is more performant than the current implementation.
+        const miterLimitVector = scaleHatVector(bisectorVector, strokeMiterLimit * s);
+        adjustedStrokeMiterLimit = hypot(miterLimitVector.x, miterLimitVector.y) / s;
+      } else {
+        adjustedStrokeMiterLimit = strokeMiterLimit
+      }
+
+      if (hypot(miterVector.x, miterVector.y) / s <= adjustedStrokeMiterLimit) {
```

Comment:
I think we can expose hypot on the Point class, naming it hypot or size

Reply:
I think it's a good one! I called it magnitude, if you prefer another name feel free to change it https://github.com/fabricjs/fabric.js/pull/8344/commits/3eec3d1c021fec636d5a39e6a560da4048c7c082